File: src/test/java/be/continuum/data/UserTest.java
package be.continuum.data;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.equalTo;
import static org.hamcrest.CoreMatchers.nullValue;
import static org.junit.Assert.assertThat;

public class UserTest {

    @Test
    public void createUser() {
        User user = new User();
        user.setSurname("John");
        user.setName("Doe");

        assertThat(user.getSurname(), equalTo("John"));
        assertThat(user.getName(), equalTo("Doe"));
        assertThat(user.getAge(), nullValue());
    }

    @Test
    public void createUserWithAge() {
        User user = createJohnDoeUser();
        user.setAge(30);

        assertThat(user.getAge(), equalTo(30));
    }

    private User createJohnDoeUser() {
        User user = new User();
        user.setSurname("John");
        user.setName("Doe");
        return user;
    }

    @Test
    public void createKotlinUser() {
        KotlinUser user = new KotlinUser();
        user.setSurname("John");
        user.setName("Doe");
        user.setAge(30);

        assertThat(user.getSurname(), equalTo("John"));
        assertThat(user.getName(), equalTo("Doe"));
        assertThat(user.getAge(), equalTo(30));

        assertThat(user.toString(), equalTo("KotlinUser(surname=John, name=Doe)"));
    }

}

File: src/test/kotlin/be/continuum/kotlin/data/UserTest.kt
package be.continuum.kotlin.data

import org.hamcrest.CoreMatchers.equalTo
import org.hamcrest.CoreMatchers.nullValue
import org.junit.Assert.assertThat
import org.junit.Test

class UserTest {

    @Test
    fun createUser() {
        val user = User(name = "Doe", surname = "John")

        assertThat(user.surname, equalTo("John"))
        assertThat(user.name, equalTo("Doe"))
        assertThat(user.age, nullValue())
    }

    @Test
    fun `Create user with age`() {
        val user = be.continuum.data.User("John", "Doe")
        user.age = 30

        assertThat(user.age, equalTo(30))
    }

}